src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -34,14 +34,27 @@ internal sealed partial class SmartRenameViewModel : INotifyPropertyChanged, IDi
     private readonly IGlobalOptionService _globalOptionService;
     private readonly IThreadingContext _threadingContext;
     private readonly IAsynchronousOperationListener _asyncListener;
-    private CancellationTokenSource? _cancellationTokenSource;
+
+    /// <summary>
+    /// Cancellation token source for <see cref="ISmartRenameSessionWrapper.GetSuggestionsAsync(ImmutableDictionary{string, ImmutableArray{ValueTuple{string, string}}}, CancellationToken)"/>.
+    /// Each call uses a new instance. Mutliple calls are allowed only if previous call failed or was canceled.
+    /// The request is canceled through
+    /// 1. <see cref="BaseViewModelPropertyChanged"/> when user types in the text box.
+    /// 2. <see cref="ToggleOrTriggerSuggestions"/> when user toggles the automatic suggestions.
+    /// 3. <see cref="Dispose"/> when the dialog is closed.
+    /// </summary>
```

Comment:
mention the threading requirements for cancelling this. (i assume it's only on hte UI thread).